### PR TITLE
fix inconsistent instance state flags which results in a busy loop

### PR DESCRIPTION
**Pull request by dirk-thomas**
Based on the investigations here: https://github.com/ros2/rclcpp/issues/192#issuecomment-183491185

It makes the test from the `issue_192` branch work for me locally.

* http://ci.ros2.org/job/ci_linux/975/
* http://ci.ros2.org/job/ci_osx/810/
* http://ci.ros2.org/job/ci_windows/1063/

### CODE REVIEW

**Comment by wjwwood** [comment]
I'd recommend adding a detailed comment about why this is the way it is. I feel that in a year we could easily look at this and think it should be another way, having forgotten why we made this change. I was curious to see if we had an interesting commit message when we added that option, but the file has been refactored and renamed, which makes it hard to follow back to the original commit.

**Comment by dirk-thomas** [comment]
I traced both locations back to its origin and they have both been added with exactly these arguments when the features of services and then later the read conditions have been added.

**Comment by wjwwood** [comment]
I did too. I think a comment in the code makes sense because even though I could trace it back, it was pretty hard and very time consuming.

**Comment by dirk-thomas** [comment]
Please feel free to suggest a comment and add it to the PR.

**Comment by dirk-thomas** [comment]
Since this change doesn't have any regressions on Linux and OS X I would like to merge this now (even with the Windows job still pending, but that will be the case for at least another hour). Any +1 votes?

**Comment by jacquelinekay** [comment]
I agree with William. I will add a comment and then approve.

**Comment by jacquelinekay** [comment]
+1

**Comment by wjwwood** [comment]
lgtm
